core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelector.java

```diff
@@ -168,47 +158,37 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
         queueSizeAdjuster.update(existingTasks, assignmentStats);
         Set<InternalNode> blockedExactNodes = new HashSet<>();
         boolean splitWaitingForAnyNode = false;
-        // splitsToBeRedistributed becomes true only when splits go through locality-based assignment
-        boolean splitsToBeRedistributed = false;
-        Set<Split> remainingSplits = new HashSet<>(splits.size());
 
         List<InternalNode> filteredNodes = filterNodes(nodeMap, includeCoordinator, ImmutableSet.of());
         ResettableRandomizedIterator<InternalNode> randomCandidates = new ResettableRandomizedIterator<>(filteredNodes);
         Set<InternalNode> schedulableNodes = new HashSet<>(filteredNodes);
 
-        // optimizedLocalScheduling enables prioritized assignment of splits to local nodes when splits contain locality information
-        if (optimizedLocalScheduling) {
-            for (Split split : splits) {
-                if (split.isRemotelyAccessible() && !split.getAddresses().isEmpty()) {
-                    List<InternalNode> candidateNodes = selectExactNodes(nodeMap, split.getAddresses(), includeCoordinator);
-
-                    Optional<InternalNode> chosenNode = candidateNodes.stream()
-                            .filter(ownerNode -> assignmentStats.getTotalSplitsWeight(ownerNode) < maxSplitsWeightPerNode && assignmentStats.getUnacknowledgedSplitCountForStage(ownerNode) < maxUnacknowledgedSplitsPerTask)
-                            .min(comparingLong(assignmentStats::getTotalSplitsWeight));
-
-                    if (chosenNode.isPresent()) {
-                        assignment.put(chosenNode.get(), split);
-                        assignmentStats.addAssignedSplit(chosenNode.get(), split.getSplitWeight());
-                        splitsToBeRedistributed = true;
-                        continue;
-                    }
-                }
-                remainingSplits.add(split);
-            }
-        }
-        else {
-            remainingSplits = splits;
-        }
-
-        for (Split split : remainingSplits) {
+        for (Split split : splits) {
             randomCandidates.reset();
 
             List<InternalNode> candidateNodes;
+            boolean exactNodes;
             if (!split.isRemotelyAccessible()) {
                 candidateNodes = selectExactNodes(nodeMap, split.getAddresses(), includeCoordinator);
+                exactNodes = true;
             }
             else {
-                candidateNodes = selectNodes(minCandidates, randomCandidates);
+                // optimizedLocalScheduling enables prioritized assignment of splits to local nodes when splits contain locality information
+                if (optimizedLocalScheduling && !split.getAddresses().isEmpty()) {
```

Comment:
Should `optimizedLocalScheduling` even be a configurable any more ? Any caching implementations which rely on this become ineffective if this config is turned off. If there is a reason for node locality to be configurable, then the connector producing the splits should itself have a config flag to disable providing host addresses.

Reply:
> Should optimizedLocalScheduling even be a configurable any more ? Any caching implementations which rely on this become ineffective if this config is turned off. If there is a reason for node locality to be configurable

Since I've did considerable policy change I prefer to keep the property for now in case it somehow breaks performance for some users. Let's deprecate it later.

Reply:
Keeping this config flag around is misleading. There is basically no "optimizedLocalScheduling" feature any more. The crux of the original feature was to find a balance between using the provided node addresses and keeping distribution of load on the workers uniform. We're deleting the code which does that and always using the same policy as `isRemotelyAccessible = false` from before (modulo the corner case of node going away).
The configuration about disabling local scheduling should move to individual connectors which should either return host addresses or not based on the flag.